### PR TITLE
Adding HappyBase batch delete().

### PR DESCRIPTION
**Pull request by dhermes**
This completes the `Batch()` implementation.

----

All that is left is finishing `Table`! w00t! (And bringing over the HappyBase system tests.)

### CODE REVIEW

File: gcloud/bigtable/happybase/test_batch.py [hunk 1]
```diff
@@ -352,7 +486,7 @@ def test_it(self):
         expected_result = [
             ['cf1', None],
             ['cf2', None],
-            ['cf3', ''],
+            ['cf3', ''],  # AGGGHHHHHHHHHH THIS IS WEIRD
```
Comment (dhermes) [inline gcloud/bigtable/happybase/test_batch.py]: @jonparrott I forgot to ask you about this when #1533 went in. Sorry.

In my old implementation I was using `'cf3::'.rstrip(':')` but I'm thinking this should fail?

Comment (jonparrott) [inline gcloud/bigtable/happybase/test_batch.py]: Is that a valid identifier?

Maybe you should use `'cf3::'.partition(':')[0]`

Comment (dhermes) [inline gcloud/bigtable/happybase/test_batch.py]: Well that's the question, should it be valid. IMO it should not. The only valid identifiers should be `fam`, `fam:` and `fam:col`.

Though [HappyBase][1] uses `rstrip`.

[1]: https://github.com/wbolster/happybase/blob/9cbd718c10a3089f234f1eac1236b631e1f8e7cd/happybase/table.py#L60-L61

Comment (jonparrott) [inline gcloud/bigtable/happybase/test_batch.py]: I'd err on the side of matching happybase's behavior.

Comment (dhermes) [inline gcloud/bigtable/happybase/test_batch.py]: Yes I think you're right. Sad face. I'll send a separate PR to restore that behavior. Does the rest of this PR LGTY? (I'm going to `git push -f` with this editorial `AGGGHHHHHHHHHH THIS IS WEIRD` remove.

Comment (jonparrott) [inline gcloud/bigtable/happybase/test_batch.py]: Yeah, LGTM.

**Pull request merged**